qiskit/quantum_info/states/densitymatrix.py

```diff
@@ -585,6 +585,33 @@ def _evolve_instruction(self, obj, qargs=None):
         vec._append_instruction(obj, qargs=qargs)
         return vec
 
+    def to_statevector(self, atol=None, rtol=None):
+        """Return a statevector from a pure density matrix.
+
+        Returns:
+            Statevector: The pure density matrix's corresponding statevector.
+                Corresponds to the eigenvector of the only non-zero eigenvalue.
+
+        Raises:
+            QiskitError: if the state is not pure.
+        """
+        if atol is None:
+            atol = self.atol
+        if rtol is None:
+            rtol = self.rtol
+
+        if not is_hermitian_matrix(self._data, atol=atol, rtol=rtol):
+            raise QiskitError("Not a valid density matrix (non-hermitian).")
+
+        evals, evecs = np.linalg.eig(self._data)
+
+        nonzero_evals = evals[abs(evals) > atol]
+        if len(nonzero_evals) != 1 or not np.isclose(abs(nonzero_evals[0]), 1,
```

Comment:
I think this should be
```suggestion
        if len(nonzero_evals) != 1 or not np.isclose(np.max(nonzero_evals), 1,
```

Reply:
If you want to allow rank-1 projector density matrices that are not-trace 1 this could also be changed to
```suggestion
        if len(nonzero_evals) != 1 or not nonzero_evals[0] > 0:
```